lib/rubocop/optimized_patterns.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module RuboCop
+  # @api private
+  module OptimizedPatterns
```

Comment:
I guess `OptimizedFilePatterns` (or just `FilePatterns`)  would be a bit more descriptive name.

Reply:
Or even `PathPatterns`! Naming is hard! :D